.pre-commit-config.yaml

```diff
@@ -127,6 +127,14 @@ repos:
     rev: 37c2513b1b8275a475a160ed2f5b044910335d5f # No release tag yet including #6 fix
     hooks:
       - id: qml_formatter
+  - repo: https://github.com/BlankSpruce/gersemi
+    rev: 0.17.1
+    hooks:
+      - id: gersemi
+  - repo: https://github.com/cmake-lint/cmake-lint.git
+    rev: 1.4.3
+    hooks:
+      - id: cmakelint
```

Comment:
How is controlled which files are checked?

Reply:
Like every other hook. The hook defines the filetype that it applies to, and pre-commit calls the `identify` library to check the filetype for ever changed file.

See https://github.com/cmake-lint/cmake-lint/blob/develop/.pre-commit-hooks.yaml#L6